packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -989,7 +989,7 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
             decorations.push(node.decorationData);
         }
         if (this.decorations.has(node.id)) {
-            decorations.push(...this.decorations.get(node.id));
+            decorations.push(...this.decorations.get(node.id) || []);
```

Comment:
You can do `get(node.id)!` as we just checked that the node was present in the map.